doc/developer/design/README.md

```diff
@@ -100,44 +102,60 @@ short.
-   small set of people who have a vested interest in the area the design
-   touches. If it's not clear who that is, ask your TL, EM, or in #eng-general.
-4. Announce that the design doc is ready for review in #eng-announce, if you
-   think it's helpful to surface the design to a larger set of people. This
-   step is optional. We have an automation that posts new ready-for-review
-   design docs to #eng-design-docs and in many cases that will be sufficient
-   to inform any interested parties.
+1. Copy the design document template to a new date-prefixed file in
+   the design/ directory of the repository you’re working in, and fill it in.
+2. Submit a pull request of your filled in design document. This
+   makes it easy for others to add written comments.
+3. Identify your key stakeholders and socialize the design document
+   with them. Typically, there is a small set of people who have a vested
+   interest in the area the design touches. If it’s not clear who your
+   stakeholders for the design document are, check in with your manager.
+4. Schedule a meeting to discuss the design document live with your
+   stakeholders. Rule of thumb is that you should give your reviewers
+   about a week to thoroughly review and engage with your document.
+5. Optional, but recommended: announce that the design document is ready
+   for review in #epd-announce. It may be useful to get extra eyes on
+   your design, but remember that this process is not intended to design
+   by committee. Use your judgment in engaging with feedback from outside
+   of your group of stakeholders.
 
 ### Iteration
 
-5. Address comments, discuss, and iterate on the document in the PR.
-6. Gather explicit approvals from relevant stakeholders. This should be the set
-   of people that you identified at step 3. or folks that noticed your design
-   document from the (optional) announcement.
+6. As you begin to get feedback on your document, address the comments
+   and answer the questions on your PR. Continue to iterate.
+7. Drive the design document meeting and attempt to gather explicit
+   approvals from your stakeholders, the set of people that you
+   identified in Step 3.
+8. If you do not get explicit approvals in the meeting, keep iterating
+   until you do. You can do this asynchronously or synchronously.
+   If you’ve gone through multiple iterations and are struggling to get
+   the required approvals, get help from your manager.
 
-### Request approval
+### Approval
 
-7. If no comments have raised new issues or if no one has asked for additional
-   time to review, merge the design document. Do _not_ assume _silent
-   consensus_: if you did not hear back from important stakeholders, don't take
-   this as approval. Make sure the relevant people have in fact seen your design
-   and had a chance to object or propose changes.
+9. Once you have gotten the required approvals, merge the design
```

Comment:
It's hard for me to reconcile

> If you did not hear back from important stakeholders, do not take this as approval. Make sure the relevant people have in fact seen your design and had a chance to object or propose changes.

with 

> Design document meetings are not meant for open-ended brainstorming, nor are they meant to act as a review body. The intention of this process is not to design by committee.

At some point I might want to merge a design doc without unanimous consensus. Are there specific guidelines that I should adhere before doing so (for example, record conflicting opinions in the design doc for posterity).

Reply:
The important bit is that relevant stakeholders were involved without formally requiring their consensus.

This part above is relevant:

> Instead, the author is responsible for driving the process to a conclusion that is satisfactory to the relevant stakeholders. Not every design document will result in complete consensus.

Regarding guidlines: I added a note in 5d62e93dde to record outcomes of the meeting. That should make it clearer whether there was consensus on a design or not.